hashes/src/sha256.rs

```diff
@@ -149,7 +149,7 @@ impl Hash {
     pub const fn const_hash(bytes: &[u8]) -> Self { Hash(Midstate::const_hash(bytes, true).bytes) }
 }
 
-/// Output of the SHA256 hash function.
+/// Unfinalized output of the SHA256 hash function.
```

Comment:
FYI hashes get finalized at the end by inputting a bunch of zeroes (you can see it in the `Hash::from_midstate` method). Midstate doesn't do this so it isn't protected against length extension attacks as mentioned in the doc.

Reply:
Ah yes, in `Hash::from_engine` - thanks man. So back in the day when I tried to move the logic in `Hash::from_engine` to `HashEngine::finalize` I had the name correct, nice :)  (I think you were away when I was doing that.)